RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate containers/Loading to `reanimated` v2 and Hooks

## Proposed changes

* Migrate to functional component
* Use `useTheme` hook
* Migrate animations to `react-native-reanimated` v2 APIs

## Issue(s)	


## How to test or reproduce

Go to the edit status screen and see if the loading that appears after saving any changes is behaving correctly.

## Screenshots

Before:


https://user-images.githubusercontent.com/77742477/165351489-05fc785c-60ab-4fd2-b246-9f2a6550ce05.mp4


After:


https://user-images.githubusercontent.com/77742477/165351552-35a6a239-011e-4004-b871-8f4ec3b1d4ea.mp4



## Types of changes




## Checklist



## Further comments
